src/lib/CombatCalc.ts

```diff
@@ -57,6 +57,24 @@ const ONE_HIT_MONSTERS: number[] = [
   11193, // Flower (A Night at the Theatre)
 ];
 
+const THRALL_DPT = 0.375;
+const THRALL_SPEED = 4;
+const thrallDistribution = new HitDistribution([new WeightedHit(0.25, [0]), new WeightedHit(0.25, [1]), new WeightedHit(0.25, [2]), new WeightedHit(0.25, [3])]);
+const zeroDistribution = new HitDistribution([new WeightedHit(1, [0])]);
+
+const getAttackingDistribution = (playerAttacks: boolean, thrallAttacks: boolean, playerDistribution: HitDistribution): HitDistribution => {
+  if (playerAttacks && thrallAttacks) {
+    return playerDistribution.zip(thrallDistribution).cumulative();
```

Comment:
Either we should not use zip here or we should cache the result of zip whenever possible. Definitely don't recompute it on every tick that both attacks happen.

zip is one of the worst-case time complexity things we can do in the ttk dist.

It's probably actually better to not zip at all, and instead just apply both dists separately since they're not correlated. ```